cro-clib/src/lib.rs

```diff
@@ -0,0 +1,263 @@
+use chain_core::init::network::init_chain_id;
+use client_common::error::ErrorKind;
+use client_common::storage::MemoryStorage;
+use client_common::SecKey;
+use client_core::types::WalletKind;
+use client_core::wallet::{DefaultWalletClient, WalletClient};
+use client_core::HDSeed;
+use client_core::Mnemonic;
+use secstr::SecUtf8;
+use std::ffi::CStr;
+use std::os::raw::c_char;
+use std::str::FromStr;
+pub const MAX_LENGTH: u32 = 1024;
+pub const MAX_STRING_LENGTH: u32 = 512;
+#[repr(C)]
+#[derive(Copy, Clone)]
+pub struct Buf {
+    pub buf: [::std::os::raw::c_uchar; 512usize],
+    pub length: ::std::os::raw::c_int,
+}
+
+#[repr(C)]
+#[derive(Copy, Clone)]
+pub struct ApiResult {
+    pub error: ::std::os::raw::c_int,
+    pub value: Buf,
+}
+
+#[repr(C)]
+#[derive(Copy, Clone)]
+pub struct HDWallet {
+    pub error: ::std::os::raw::c_int,
+    pub name: Buf,
+    pub value: Buf,
+    pub mnemonics: Buf,
+    pub viewkey: Buf,
+    pub seed: Buf,
+    pub enckey: Buf,
```

Comment:
I'll repeat the same feedback -- these are different types, reflect that in the binding.
For example, for view key, just use `secp256k1::ffi::PublicKey`

Reply:
Maybe make some fields opaque(`mnemonics`, `viewkey`, `seed`, `enckey`), expose related functionality through APIs.
What's the meaning of `value` field?

Reply:
> What's the meaning of value field?
that's a mystery to me too -- in the code, it only copied the same value as `mnemonics`

Reply:
it's for general result value,
int error;
char* value;
but i'll change it to type for each function